lib/puppet/pops/migration/migration_checker.rb

```diff
@@ -6,6 +6,10 @@ class Puppet::Pops::Migration::MigrationChecker
   def initialize()
   end
 
+  def self.singleton
+    @@null_checker ||= self.new
```

Comment:
as I understand it, @@name makes it shared among subclasses (even if done at class level) as opposed to @name in a class - I have been bit by that in the past. Here I don't think it matters.